src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,130 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] values;
+
+    public MyHashMap() {
+        values = new Node[DEFAULT_SIZE];
+        threshold = (int) (DEFAULT_SIZE * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
+        if (isEmpty()) {
+            return null;
+        }
+        int hash = hash(key);
+        int index = hash % values.length;
+        Node<K, V> currentNode = values[index];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public V remove(K key) {
+        int hash = hash(key);
+        int index = hash % values.length;
+        Node<K, V> currentNode = values[index];
+        if (currentNode != null && (Objects.equals(key, currentNode.key))) {
+            values[index] = currentNode.next;
+            size--;
+            return currentNode.value;
+        }
+        while (currentNode != null) {
+            Node<K, V> nextNode = currentNode.next;
+            if (nextNode != null && (Objects.equals(key, nextNode.key))) {
+                currentNode.next = nextNode.next;
+                size--;
+                return nextNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private void putValue(int hash, K key, V value) {
```

Comment:
It looks complicated, try simplifying it:
```
    public void put(K key, V value) {
        if (size > table.length * LOAD_FACTOR) {
            resize();
        }
        int index = getIndex(key);
        Node<K, V> node = table[index];
        while (node != null) {
            if (Objects.equals(node.key, key)) {
                node.value = value;
                return;
            }
            node = node.next;
        }
        Node<K, V> nodeNew = new Node<>(key, value, table[index]);
        table[index] = nodeNew;
        size++;
    }
```

Reply:
Thanks. Indeed, it is possible to insert a new node at the beginning.